grc/gui/ParamWidgets.py

```diff
@@ -334,3 +334,47 @@ def _handle_clicked(self, widget=None):
             self._editing_callback()
             self._apply_change()
         file_dialog.destroy()  # destroy the dialog
+
+class DirectoryParam(EntryParam):
+    """Provide an entry box for a directory and a button to browse for it."""
+
+    def __init__(self, *args, **kwargs):
+        EntryParam.__init__(self, *args, **kwargs)
+        self._open_button = Gtk.Button(label='...')
+        self._open_button.connect('clicked', self._handle_clicked)
+        self.pack_start(self._open_button, False, False, 0)
+
+    def _handle_clicked(self, widget=None):
+        """
+        Open the directory selector, when the button is clicked.
+        On success, update the entry.
+        """
+        dirname = self.param.is_valid() and self.param.get_evaluated() or '' # Obtain current value with fallback
+        
+        if not os.path.exists(dirname): # Check if directory exists, if not fall back to workdir
```

Comment:
may I suggest to use `os.path.isdir`. I am not quite sure if that "sees" through symlinks. So that should be tested.

Reply:
According to the documentation [os.path.isdir](https://docs.python.org/3.8/library/os.path.html#os.path.isdir) symlinks are handled. Thanks for the suggestion!